### PR TITLE
MB-61421: Interfaces for supporting online backup of an index

### PR DESCRIPTION
**Pull request by CascadingRadium**
- Added CopyIndex interface extending Index with a getter for CopyReader for online index backups.
- Introduced CopyReader interface extending IndexReader with a CopyTo method for online index copy/backup.

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -225,3 +225,25 @@ type IndexBuilder interface {
...
+// of the index.
+type IndexCopyable interface {
+	// CopyTo creates a fully functional copy of the index at the
+	// specified destination directory implementation.
+	CopyTo(d Directory) error
+}
+
+type ReaderCopyable interface {
+	CopyableReader() CopyReader
+}
```
Comment (abhinavdangeti) [inline index.go:239]: Naming is quite funky :), also these need not really be exported - and should be relocated into bleve, alongside the IndexCopyable.

**Review by abhinavdangeti — CHANGES_REQUESTED** [review]
@CascadingRadium Let's move this code back into `bleve` where it originally resided. There's no need for introducing this here as bleve is the only caller.

Also let's make those that don't need exporting private.

File: index.go (hunk 1 above)
Comment (CascadingRadium) [reply index.go:239]: ok

**Comment by CascadingRadium** [comment]
i have kept the existing interface as is in bleve and added the 2 new interfaces here 
- CopyIndex : extension of Index interface (for getting the CopyReader)
- CopyReader : extension of IndexReader interface (for supporting CopyTo and Close methods)

**Review by abhinavdangeti — APPROVED** [review]

**Pull request merged**